HIP/hip-991.md

```diff
@@ -0,0 +1,221 @@
+* **Ease of Monetization:** Developers and businesses find it challenging to charge users for submitting information to a topic due to the lack of a native fee structure. A built-in payment system would streamline monetization, allowing developers to easily charge for topic submissions, which can enhance the financial viability of their applications.
+* **Resource Allocation:** The current low fee of $0.0001 per message makes it difficult for topic operators to manage and predict the economic impact of high-traffic topics. Users can suffer from spam and unexpected messages that operators have to handle, potentially degrading the quality of service.
+* **Token and HBAR Utilization:** There is an opportunity to expand the utility of various tokens within the Hedera ecosystem. By allowing fees to be paid in tokens other than HBAR, such as those required by DAOs, users can leverage their token holdings for topic submissions, enhancing the ecosystem's flexibility and user engagement.
+
+---
+
+**Feature Summary:**
+
+* Introduces an optional fee for submitting messages to a topic. When fixed fees are enabled, HBAR or FT payments are required for message submissions, with exemptions for holders of a submit key. Submit keys are not required for topics with fees.
+* Introduces a Fee Schedule Key for topic creation. This key can set the fee for message submissions, with exemptions for holders of a submit key.
+* Fees can be distributed similarly to fixed fees on the HTS, supporting multiple wallets, percentage distributions, and Fungible token in addition to HBAR.
+* Cannot add Key after creation - similar to tokens.
+
+**Benefits:**
+
+* Provides predictable revenue per message for topic operators.
+* Enables new products and business models on HCS.
+* Increases node revenue for network fees tied to new HCS transactions/topic creations with fixed fees.
+* Enhances the utility of HCS within Hedera.
+* Increases revenue streams for node operators.
+* Enables simple facilitation of holding and spending tokens to document new information on a topic.
+
+**Related Requests:** This enhancement is in response to community feedback (Tier Bot) requesting more versatile and predictable economic models within topic IDs on the Hedera network.
+
+## **Specification**
+
+### **Conceptual Overview**
+
+We propose adding a fixed fee mechanism to the Hedera Consensus Service (HCS) for topic messages, similar to the custom fee structures in the Hedera Token Service (HTS). This will involve creating new Protobuf messages and modifying existing ones to accommodate fixed fees for topics.
+
+### **New Protobuf Messages**
+
+### **FixedFee**
+
+The `FixedFee` message represents a fixed fee charged for each message submission to a topic.
+
+```
+message FixedFee {
+  // The number of units to assess as a fee
+  int64 amount = 1;
+  // The denomination of the fee; taken as hbar if left unset
+  TokenID denominating_token_id = 2;
```

Comment:
I wonder if we should call this "currency" or just "denomination" since the "token_id" at the end is a bit redundant.